frontend/pages/sites/$siteId/storage/LocationBar.jsx

```diff
@@ -2,18 +2,22 @@ import React, { useMemo } from 'react';
 import PropTypes from 'prop-types';
 import { IconGlobe } from '@shared/icons';
 
-const LocationBar = ({ path, storageRoot, onNavigate }) => {
+const LocationBar = ({ path, storageRoot, onNavigate, trailingSlash = true }) => {
   // Only trim leading slashes, keep trailing
   const cleanedPath = path.replace(/^\/+/, '');
   const segments = cleanedPath ? cleanedPath.split('/').filter(Boolean) : [];
   // Show storageRoot only if it's the current or parent folder
   const showDomain = segments.length <= 1;
-  const grandparentPath = segments.slice(0, segments.length - 1).join('/') + '/';
+  const grandparentPath = segments.slice(0, -2).join('/') + '/';
 
   const displayedBreadcrumbs = useMemo(() => {
+    const trailer = trailingSlash ? '/' : '';
```

Comment:
This part should make sure that what's up there is a folder first:
![Screenshot 2025-03-28 at 2 45 02 PM](https://github.com/user-attachments/assets/d993c32e-90b3-4327-baeb-f097d0a01704)

Reply:
Fixed for files in the root.